momepy/preprocessing.py

```diff
@@ -247,14 +248,16 @@ def remove_false_nodes(gdf):
             ),
             lines=False,
         )
+
         loops = combined[combined.is_ring]
+
         node_ix, loop_ix = loops.sindex.query(nodes.geometry, predicate="intersects")
         for ix in np.unique(loop_ix):
             loop_geom = loops.geometry.iloc[ix]
             target_nodes = nodes.geometry.iloc[node_ix[loop_ix == ix]]
             if len(target_nodes) == 2:
                 node_coords = shapely.get_coordinates(target_nodes)
-                coords = np.array(loop_geom.coords)
+                coords = np.array(force_2d(loop_geom).coords)
```

Comment:
```suggestion
                coords = shapely.get_coordinates(loop_geom)
```
No need to create a new (2D) geometry and then fetch its coordinates.